testing/rdiffbackupdeletetest.py

```diff
@@ -0,0 +1,144 @@
+import os
+import sys
+
+from distutils import spawn
+import subprocess
+import unittest
+
+PY2 = sys.version_info < (3,)
+PY3 = sys.version_info > (3,)
+
+# Working directories
+old_test_dir = os.path.join(os.path.dirname(os.getcwd()), 'rdiff-backup_testfiles')
+
+
+# Lookup for rdiff-backup-delete location.
+def rdiff_backup_delete(to_delete=None, extra_args=[], expected_ret_val=0, expected_output=None):
+    bin = spawn.find_executable(u"rdiff-backup-delete")
+    assert bin, "can't find rdiff-backup-delete"
+    cmdargs = [bin.encode('utf8')]
+    if extra_args:
+        cmdargs.extend(extra_args)
+    if to_delete:
+        cmdargs.append(to_delete)
+    cmdline = b" ".join(cmdargs)
+    print("Executing: %r" % (cmdline,))
+    p = subprocess.Popen(cmdargs, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    output, error = p.communicate()
+    ret_val = p.poll()
+    if expected_ret_val is not None:
+        assert ret_val == expected_ret_val, "Return code %d of command `%s` doesn't matches expected value: %d.\n%s\n%s" % \
+            (ret_val, cmdline, expected_ret_val, output, error)
+    if expected_output is not None:
+        assert expected_output in output or expected_output in error, "Output %s %s of command `%s` doesn't matches expected output:\n`%s`" % \
+            (output, error, cmdline, expected_output)
+
+
+def rdiff_backup_verify(backup_dir):
+    assert backup_dir
+    assert os.path.isdir(backup_dir)
+
+    # Lookup rdiff-backup executable.
+    bin = spawn.find_executable("rdiff-backup")
+    assert bin, "can't find rdiff-backup"
+
+    # Build the command line
+    cmdargs = [bin.encode('utf8'), b'--verify', backup_dir]
+    cmdline = b" ".join(cmdargs)
+    print("Executing: %r" % (cmdline,))
+    DEVNULL = open(os.devnull, 'wb')
+    try:
+        p = subprocess.Popen(cmdargs, stdout=DEVNULL, stderr=DEVNULL)
+        assert p.wait() == 0, 'rdiff-backup verify failed'
+    finally:
+        DEVNULL.close()
+
+
+class RdiffBackupDeleteTest(unittest.TestCase):
+
+    repo = ""
+
+    def _copy_repo(self, reponame):
+        # Copy the required repo to a temporary location.
+        # We need to use os command line to properly copy and delete special files.
+        self.repo = b"/tmp/deletetest"

```

Comment:
Don't use `/tmp` but variable `abs_test_dir` (from commontests.py) as basis for all tests.

Reply:
Fixed